packages/theme/components/StoreSwitcher/__tests__/StoresModal.spec.ts

```diff
@@ -0,0 +1,81 @@
+import userEvent from '@testing-library/user-event';
+import { render } from '~/test-utils';
+import StoresModal from '~/components/StoreSwitcher/StoresModal.vue';
+import { useStoreMock } from '~/test-utils/mocks/useStore';
+import { useStore } from '~/composables';
+import { availableStoresMock } from '~/components/StoreSwitcher/__tests__/availableStores.mock';
+
+jest.mock('~/composables', () => {
+  const originalModule = jest.requireActual('~/composables');
+  return {
+    ...originalModule,
+    useStore: jest.fn(),
+  };
+});
+
+describe('<StoresModal>', () => {
+  it('does not render when closed', () => {
+    (useStore as jest.Mock).mockReturnValue(useStoreMock);
+    const { queryByRole } = render(StoresModal, {
+      props: {
+        isLangModalOpen: false,
+      },
+    });
+
+    expect(queryByRole('heading', { name: /change store/i })).toBeNull();
+  });
+
+  it('emits "closeModal" event on close', async () => {
+    const user = userEvent.setup();
+
+    (useStore as jest.Mock).mockReturnValue(useStoreMock);
+    const { getAllByRole, emitted } = render(StoresModal, {
+      props: {
+        isLangModalOpen: true,
+      },
+    });
+
+    const closeBtn = getAllByRole('button', { name: /close/i })[0];
+
+    await user.click(closeBtn);
+
+    expect(emitted()).toHaveProperty('closeModal');
+  });
+
+  it('renders list of available stores', () => {
+    useStoreMock.stores.value = availableStoresMock;
+
+    (useStore as jest.Mock).mockReturnValue(useStoreMock);
+
+    const { queryByRole, queryByText } = render(StoresModal, {
+      props: {
+        isLangModalOpen: true,
+        storeConfig: { store_code: 'default' },
+      },
+    });
+
+    expect(queryByRole('heading', { name: /change store/i })).toBeTruthy();
+    expect(queryByText(/default store view/i)).toBeTruthy();
+    expect(queryByText(/german/i)).toBeTruthy();
```

Comment:
queryBy* is more used to check for the absence of elements.

You can just use getByRole and have it be an implicit assertion

```suggestion
    getByRole('heading', { name: /change store/i }));
   // and so on
```

Reply:
Or I can use query and explicit assertion. Thank you ;)

Reply:
I do not like this change but reviewers think differently.